### PR TITLE
simplify the code

### PR DESCRIPTION
**Pull request by CosmosShadow**

### CODE REVIEW

**Review by soumith — APPROVED** [review]

File: torch/lib/TH/generic/THTensorMath.c [hunk 1]
```diff
@@ -3203,10 +3200,7 @@ void THTensor_(logspace)(THTensor *r_, real a, real b, long n)
   }
 
   if(n == 1) {
-    TH_TENSOR_APPLY(real, r_,
-        *r__data = TH_MATH_NAME(pow)(10.0, a);
-        i++;
-        );
+    THTensor_(set1d)(r_, 0, TH_MATH_NAME(pow)(10.0, a));
```
Comment (apaszke) [inline torch/lib/TH/generic/THTensorMath.c]: Note that `set1d` does in-bound checks that are unnecessary

Comment (soumith) [reply torch/lib/TH/generic/THTensorMath.c]: ya but this is literally for n==1 in linspace and logspace. i think it's fine.

**Comment by soumith** [comment]
Thanks @CosmosShadow !

**Pull request merged**